fixed-highlight of bing-streetview and kartaview

draunger opened the pull request:
# description :
fixes the highlight of sequence on hover 
# after the fix :
- bing street view sequence :
![image](https://github.com/user-attachments/assets/e86d769e-9237-4af9-8a66-c59df7870e62)
- kartaview street view sequence:
![image](https://github.com/user-attachments/assets/64480376-3a72-4c6b-a746-157fb4034a61)
# solves:
https://github.com/openstreetmap/iD/issues/10755

draunger committed:
fixed-highlight-bing-streetview-and-kartaview

draunger commented:
@tyrasd i have added the css regarding the kartaview and bing you can check and comment if you seem any problem ..

tordans commented on css/60_photos.css:
```diff
@@ -140,7 +140,17 @@ li.list-item-photos.active:after {
 .viewfield-group.hovered * {
     fill: #eebb00 !important;
```
Can we move this down into the two new sections? That would be a slight duplication but should allow to get rid of the !important and reduce complexity.

draunger replied:
@tordans can you please elaborate what you wanted me to do ?

draunger replied:
![image](https://github.com/user-attachments/assets/79fe8bbb-fe85-45d8-8a92-4140b76216c0)
@tordans  are you wanted to me to do this ?

tordans replied:
Yes, that is what I meant. Does this  still render as intended?

draunger replied:
@tordans yes , this works

tyrasd reviewed (changes requested):
There is still something not working for the _selected photo_ marker (and viewfield cone) of the Bing Streetside service (it should be much brighter):

![image](https://github.com/user-attachments/assets/673ed671-25f8-4422-9d88-0c1a59208bfb)

(the css class for this state should be `.currentView`).


And as [mentioned](https://github.com/openstreetmap/iD/issues/10755#issuecomment-2647917329) earlier, we should try to keep the differences between the services to a minimum (except for the different base color of course):

Panoramax "current view" markers still have the white border on the cone:

![image](https://github.com/user-attachments/assets/ac8b76ef-aaae-4875-9598-308905abe568)

And Mapilio's "current view" markers has only white borders:

![image](https://github.com/user-attachments/assets/cf851b06-17c5-4e82-bf49-1929d74a8b3b)

draunger committed:
added the markers

draunger commented:
@tyrasd i have added the markers and rest of thing which are left out ..

tyrasd commented on css/60_photos.css:
```diff
@@ -311,6 +328,13 @@ li.list-item-photos.list-item-mapilio.active:after {
+    stroke-width: 1;
+    stroke-opacity: 1;
+    fill-opacity: 0.4;
+}/* panoramax Image Layer */
```
```suggestion
}
```

tyrasd commented on css/60_photos.css:
```diff
@@ -343,6 +367,13 @@ li.list-item-photos.list-item-mapilio.active:after {
     overflow: hidden;
 }
 
+.layer-panoramax .viewfield-group.currentView * {
```
please move this down below the `/* panoramax Image Layer */` section header

tyrasd reviewed (commented):
Wouldn't it be better to consolidate the common properties of the different `.currentView` styles into a single rule for all services instead of duplicating them n-fold?

draunger commented:
> Wouldn't it be better to consolidate the common properties of the different `.currentView` styles into a single rule for all services instead of duplicating them n-fold?

yes you are right

draunger committed:
consolidate the commonthings

draunger commented:
@tyrasd i have changes the things as you said ..

tyrasd committed:
drop some redundant css rules, sort more logically and clean up

tyrasd reviewed (approved):
Ok. I just pushed a commit to this PR where I fixed some remaining redundant rules and cleaned up the css a bit further.

Can you please check if this is still all working as expected @draunger? If yes, we can merge this.

draunger commented:
> Ok. I just pushed a commit to this PR where I fixed some remaining redundant rules and cleaned up the css a bit further.
> 
> Can you please check if this is still all working as expected @draunger? If yes, we can merge this.

@tyrasd yes, the things are working as expected i have checked all :)

tyrasd committed:
Merge branch 'develop' into Kartaview-sequence-hover-highlight-is-too-dim

tyrasd committed:
add to changelog

tyrasd committed:
Merge branch 'develop' into Kartaview-sequence-hover-highlight-is-too-dim
